gdpr_cookies.module

```diff
@@ -605,3 +606,23 @@ function gdpr_cookies_autoload_info() {
     'ThirdPartyServiceEntity' => 'src/Entity/ThirdPartyServiceEntity.php',
   );
 }
+
+/**
+ * Implements hook_filter_info_alter().
+ */
+function gdpr_cookies_filter_info_alter(&$info) {
+  if (isset($info['filter_html'])) {
+    $info['filter_html']['process callback'] = 'gdpr_cookies_filter_process';
+  }
+}
+
+/**
+ * Callback for removing `srcdoc` attributes.
+ */
+function gdpr_cookies_filter_process($text, $filter, $format, $langcode, $cache, $cache_id) {
+  // Remove the whole srcdoc attribute.
+  $pattern = '/<([^>]*)\s+srcdoc\s*=\s*(["\'])[^"\']*\2([^>]*)>/i';
```

Comment:
I feel like this could be evaded, for example like this:
```html
<a class="youtube_player" srcdoc='<img src="x" onerror=alert()>'>Test</a>
```

(CKEditor will reformat the quotes in this, but it is possible to bypass it by disabling JavaScript.)

I think using the PHP DOM extension would be more robust.

Reply:
```php
/**
 * Removes srcdoc attributes from HTML content using the DOM extension.
 *
 * @param string $html
 *   The HTML content to process.
 *
 * @return string
 *   The processed HTML with srcdoc attributes removed.
 */
function gdpr_cookies_filter_process($html) {
  // Create a new DOM document
  $dom = new DOMDocument();
  $dom->preserveWhiteSpace = true;
  
  // Use XPath to find all elements with srcdoc attribute
  $xpath = new DOMXPath($dom);
  $elements = $xpath->query('//*[@srcdoc]');
  
  // Remove the srcdoc attribute from all matching elements
  foreach ($elements as $element) {
    $element->removeAttribute('srcdoc');
  } 

  return $dom->saveHTML();
}
```

Reply:
Yes, it works as long as I have "Limit allowed HTML tags" enabled on the text format.